crates/radix-engine-toolkit/src/transaction_types/traverser/types/general_subintent.rs

```diff
@@ -0,0 +1,183 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use radix_transactions::manifest::*;
+use radix_transactions::prelude::*;
+use scrypto::prelude::*;
+
+use radix_engine_interface::blueprints::account::*;
+
+use crate::transaction_types::*;
+
+pub struct GeneralSubintentDetector {
+    is_valid: bool,
+    is_yield_to_parent_present: bool,
+}
+
+impl GeneralSubintentDetector {
+    pub fn is_valid(&self) -> bool {
+        self.is_valid && self.is_yield_to_parent_present
+    }
+
+    pub fn output(self) -> Option<()> {
+        if self.is_valid() { Some(()) } else { None }
+    }
+}
+
+impl StaticAnalysisCallback for GeneralSubintentDetector {
+    fn on_finish(&mut self, instructions_count: usize) {
+        if instructions_count == 0 {
+            self.is_valid = false
+        }
+    }
+
+    fn on_instruction(&mut self, instruction: &InstructionV2, _: usize) {
+        // Control whether or not this is allowed or not based on:
+        // 1. Whether the instruction is allowed.
+        // 2. Whether the instruction contents are allowed.
+        self.is_valid &= match instruction {
```

Comment:
MINOR/OPTIONAL: I wonder if we should be more explicit here to avoid duplication? e.g. have a `is_instruction_permitted()` method in `General` which we could use here, via e.g.
```rust
match instruction {
    AnyInstruction::VerifyParent(..) => true,
    AnyInstruction::YieldToParent(..) => {
        self.is_yield_to_parent_present = true;
        true
    }
    instruction => GeneralIntentDetector::is_instruction_permitted(instruction),
}
```